src/Utilities/FlowAnalysis/FlowAnalysis/Analysis/TaintedDataAnalysis/TaintedDataAnalysis.TaintedDataOperationVisitor.cs

```diff
@@ -232,13 +232,23 @@ public override TaintedDataAbstractValue VisitObjectCreation(IObjectCreationOper
                     ProcessTaintedDataEnteringInvocationOrCreation(method, taintedArguments, originalOperation);
                 }
 
+                bool isSanitizingMethod = false;
                 PooledHashSet<string>? taintedTargets = null;
                 PooledHashSet<(string, string)>? taintedParameterPairs = null;
+                PooledHashSet<(string, string)>? sanitizedParameterPairs = null;
                 try
                 {
-                    if (this.IsSanitizingMethod(method))
+                    var taintedParameterNames = visitedArguments
+                            .Where(s => this.GetCachedAbstractValue(s).Kind == TaintedDataAbstractValueKind.Tainted)
+                            .Select(s => s.Parameter.Name); // keep enumerable, don't cache it here
```

Comment:
Would rather just cache the `.ToImmutableArray()`, since it's gonna be realized twice.

Bonus points for changing `IsSanitizingMethod()` and `IsSourceTransferMethod()` to take an `ISet<string> taintedParameterNames`, and using a `PooledHashSet<string>` instead of an `ImmutableArray<string>` for `taintedParameterNames`.

Reply:
It is made on purpose. If resolved once it breaks `Test_Source_ASCIIEncodingGetBytes_WithStringAndInt32AndInt32AndByteArrayAndInt32Parameters_WithConstantString_Diagnostic` and `Test_ASCIIEncodingGetBytesWithStringAndInt32AndInt32AndByteArrayAndInt32Parameters_CreateEncryptor_Diagnostic` tests. I'll update the comment.

Reply:
On second thought, I have decided to rewrite with caching.

Reply:
Oh, the set of tainted parameters can change. Got it. Your cached approach looks fine.